src/main/java/hudson/tasks/junit/TestResultAction.java

```diff
@@ -86,8 +88,14 @@ public TestResultAction(TestResult result, BuildListener listener) {
         this((Run) null, result, listener);
     }
 
+    @SuppressWarnings("deprecation")
     @Override public Collection<? extends Action> getProjectActions() {
-		return Collections.<Action>singleton(new TestResultProjectAction(run.getParent()));
+        Job<?,?> job = run.getParent();
+        if (/* getAction(Class) produces a StackOverflowError */!Util.filter(job.getActions(), TestResultProjectAction.class).isEmpty()) {
+            // JENKINS-26077: someone like XUnitPublisher already added one
+            return Collections.emptySet();
+        }
+		return Collections.singleton(new TestResultProjectAction(job));
```

Comment:
It makes sense to finally get rid of tabs, because the string has been modified in any case